core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -508,12 +508,12 @@ abstract class AbstractControllerBrokerRequestBatch(config: KafkaConfig,
   /** Send UpdateMetadataRequest to the given brokers for the given partitions and partitions that are being deleted */
   def addUpdateMetadataRequestForBrokers(brokerIds: Seq[Int],
                                          partitions: collection.Set[TopicPartition]): Unit = {
-
+    val controllerContextSnapshot = ControllerContextSnapshot(controllerContext)
```

Comment:
What does this do?

Reply:
This caches the computed fields inside ControllerContextSnapshot so that they don't need to be repeatedly computed.